enclave_manager/avalon_enclave_manager/base_enclave_manager.py

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import sys
+import zmq
+from abc import ABC, abstractmethod
+
+import avalon_enclave_manager.avalon_enclave_info as enclave_info
+from database import connector
+from avalon_enclave_manager.worker_kv_delegate import WorkerKVDelegate
+from avalon_enclave_manager.work_order_kv_delegate import WorkOrderKVDelegate
+from avalon_sdk.worker.worker_details import WorkerStatus, WorkerType
+
+logger = logging.getLogger(__name__)
+
+
+class EnclaveManager(ABC):
+    """
+    Abstract base class for Enclave Manager
+    """
+
+    def __init__(self, config):
+
+        super().__init__()
+
+        self._config = config
+        signup_data, measurements = self._setup_enclave()
+        self._kv_helper = self._connect_to_kv_store()
+        self._worker_kv_delegate = WorkerKVDelegate(self._kv_helper)
+        self._wo_kv_delegate = WorkOrderKVDelegate(self._kv_helper)
+        self.enclave_data = signup_data
+        self.sealed_data = signup_data.sealed_data
+        self.verifying_key = signup_data.verifying_key
+        self.encryption_key = signup_data.encryption_key
+
+        # TODO: EncryptionKeyNonce and EncryptionKeySignature are hardcoded
```

Comment:
Remove Encryption key signature from TODO comment, as we are already generating it as part of signup info.